compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -5178,6 +5178,34 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             _ => {}
         }
     }
+
+    pub fn suggest_remove_brackets_from_range(
+        &self,
+        err: &mut Diag<'_>,
+        main_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        leaf_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        span: Span,
+    ) {
+        if self.tcx.is_diagnostic_item(sym::RangeBounds, leaf_trait_predicate.def_id())
+            && let ty::Array(inner_type, arr_size) =
+                main_trait_predicate.self_ty().skip_binder().kind()
+            && let ty::Adt(adt_def, _) = inner_type.kind()
+            && self.tcx.is_range(adt_def.did())
+            && let Some(1) = arr_size.try_to_target_usize(self.tcx)
+        {
+            if let Ok(snip) = self.tcx.sess.source_map().span_to_snippet(span)
+                && snip.starts_with('[')
+                && snip.ends_with(']')
```

Comment:
> Right now `suggest_remove_brackets_from_range` only works for inline ranges - for that it checks if snippet starts and ends with `[`, `]`. It this the right solution?

Well, it does make it considerably easier to construct the suggestion, and there is precedence for it in `suggest_block_to_brackets`, so I think it's fine.